### PR TITLE
Allow bonus levels to be accessed by level name

### PR DESCRIPTION
**Pull request by bethanyaconnor**
We're currently refactoring the i18n pipeline to use puzzle URLs as keys to provide better context for translators. In this process, we found that we could not determine the production URL for lesson extras on the server we run the i18n sync on. This change allows us to use level name instead of id to direct to the lesson extras.

Note that bonus levels can still be accessed by ScriptLevel id and that will take precedence. This change would allow us to determine the URL for bonus levels on any environment, which would unblock work on the i18n pipeline that keys strings by the URL.

### CODE REVIEW

File: dashboard/app/controllers/script_levels_controller.rb [hunk 1]
```diff
@@ -204,6 +204,18 @@ def stage_extras
...
+      @script = Script.get_from_cache(params[:script_id])
+      @stage = @script.stage_by_relative_position(params[:stage_position].to_i)
+      @script_level = @level.script_levels.find_by_script_id(@script.id)
+      if @script_level
+        @game = @level.game
+        present_level
+        return
+      end
+    end
+
```
Comment (Erin007) [inline dashboard/app/controllers/script_levels_controller.rb]: Is there a way to add test coverage for this?

**Review by Hamms — COMMENTED** [review]
+1 for test coverage!

I'd love to see these two blocks get combined a bit more than they currently are. Maybe something like

```ruby
@script = Script.get_from_cache(params[:script_id])
@stage = @script.stage_by_relative_position(params[:stage_position].to_i)

if params[:id]
  @script_level = Script.cache_find_script_level params[:id]
  @level = @script_level.level
elsif params[:level_name]
  @level = Level.find_by_name params[:level_name]
  @script_level = @level.script_levels.find_by_script_id(@script.id)
end

if @level
  @game = @level.game
  present_level
  return
end
```

**Comment by bethanyaconnor** [comment]
Thanks all! I added a test and combined that bit of code a bit.

**Review by Erin007 — APPROVED** [review]
Changes look great! Thanks for adding tests.

**Pull request merged**